UnityProject/Assets/Scripts/Chemistry/ReagentContainer.cs

```diff
@@ -68,7 +71,7 @@ public class ReagentContainer : MonoBehaviour {
 		double DivideAmount = Using / CurrentCapacity;
 
 		Dictionary<string, float> Transfering = new Dictionary<string, float> ();
-		Dictionary<string, float> BrokenCS = new Dictionary<string, float> (Contents); 
+		Dictionary<string, float> BrokenCS = new Dictionary<string, float> (Contents);  //Because apparently am changing the order just by modifying the values That are tied with keys
```

Comment:
If order is important you could use an OrderedDictionary instead: https://docs.microsoft.com/en-us/dotnet/api/system.collections.specialized.ordereddictionary?redirectedfrom=MSDN&view=netcore-2.2

Reply:
It doesn't matter too much About the order,  It's mainly because stuff like "Contents [Chemical.Key] = 0;" Apparently messes up the for loop so That's why it's there, I could change the Contents to an OrderedDictionary then Change the loop to use that or Does that not allow me to change Contents [Chemical.Key] = 0; to  Contents.Remove(Chemical.Key)?

Reply:
yeah, That's how it set up currently But setting the value to zero still causes the loop to break

Reply:
That's why it's called BrokenCS  Because it makes no sense Why should cause the dictionary to change length/break a for loop